index.js

```diff
@@ -81,13 +80,18 @@ class ClinicDoctor extends events.EventEmitter {
       env: Object.assign({}, process.env, customEnv)
     })
 
-    if (this.detectPort) {
-      proc.stdio[3].once('data', (data) => {
+    // Use fd 3 for both Port and Source Warning
+    proc.stdio[3].once('data', (data) => {
+      if (this.detectPort) {
```

Comment:
If `detectPort` is true, and a source warning is emitted, this will run
```
this.emit('port', Number('source_warning'), proc, () => {})
```
which is not quite as intended!

Reply:
Ah yes. I can see the issue.